tmt/base.py

```diff
@@ -4462,6 +4467,9 @@ def save(self) -> None:
         )
         self.write(Path('run.yaml'), tmt.utils.dict_to_yaml(data.to_serialized()))
 
+        if self.recipe_builder is not None:
+            self.recipe_builder.set_run(data, self.fmf_context)
```

Comment:
Could we maybe have a single `save` function and we pass it the current part that it needs to save? Having individual functions makes it a bit hard to check the order in which each of these will be executed.

Reply:
If it was done step-by-step, then not having multiple methods that it is interfacing through would be nice in order to avoid figuring out what each part is about. But with it being a single function, then it resolves that issue.